### PR TITLE
Follow up PR for eventfd shim

### PR DESCRIPTION
**Pull request by tiif**
Follow up of https://github.com/rust-lang/miri/pull/3650

### CODE REVIEW

File: src/shims/unix/linux/eventfd.rs [hunk 1]
```diff
@@ -9,8 +10,8 @@ use crate::{concurrency::VClock, *};
 
 use self::shims::unix::fd::FileDescriptor;
 
-/// Minimum size of u8 array to hold u64 value.
-const U64_MIN_ARRAY_SIZE: usize = 8;
+// Size of u8 array to hold u64.
+const U64_ARRAY_SIZE: usize = mem::size_of::<u64>();
```
Comment (tiif) [inline src/shims/unix/linux/eventfd.rs:14]: Is it ok to leave this as const? ``bytes.first_chunk_mut::<U64_ARRAY_SIZE>()`` can only take in const for ``U64_ARRAY_SIZE``.

Comment (RalfJung) [reply src/shims/unix/linux/eventfd.rs:14]: Yeah a const is fine.

File: src/shims/unix/linux/eventfd.rs [hunk 2]
```diff
@@ -9,8 +10,8 @@ use crate::{concurrency::VClock, *};
 
 use self::shims::unix::fd::FileDescriptor;
 
-/// Minimum size of u8 array to hold u64 value.
-const U64_MIN_ARRAY_SIZE: usize = 8;
+// Size of u8 array to hold u64.
```
Comment (RalfJung) [inline src/shims/unix/linux/eventfd.rs:13]:
```suggestion
// We'll only do reads and writes in chunks of size u64.
```

File: src/shims/unix/linux/eventfd.rs [hunk 3]
```diff
@@ -63,15 +64,15 @@ impl FileDescription for Event {
                 throw_unsup_format!("eventfd: blocking is unsupported");
             }
         } else {
-            // Prevent false alarm in data race detection when doing synchronisation via eventfd.
+            // Future `read` calls will synchronize with this write, so update the FD clock.
```
Comment (RalfJung) [inline src/shims/unix/linux/eventfd.rs:67]: This is a "read", I think you mixed up the 2 comments?
```suggestion
            // Synchronize with past `write` calls.
```

File: src/shims/unix/linux/eventfd.rs [hunk 4]
```diff
@@ -110,8 +111,10 @@ impl FileDescription for Event {
         // Else, block.
         match self.counter.checked_add(num) {
             Some(new_count @ 0..=MAX_COUNTER) => {
-                // Prevent false alarm in data race detection when doing synchronisation via eventfd.
-                self.clock.join(&ecx.release_clock().unwrap());
+                // Synchronize with all prior `write` calls to this FD.
```
Comment (RalfJung) [inline src/shims/unix/linux/eventfd.rs:114]:
```suggestion
                // Future `read` calls will synchronize with this write, so update the FD clock.
```

File: src/shims/unix/linux/eventfd.rs [hunk 5]
```diff
@@ -163,19 +166,21 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
...
             flags &= !efd_nonblock;
             is_nonblock = true;
         }
         if flags != 0 {
-            let einval = this.eval_libc("EINVAL");
-            this.set_last_error(einval)?;
-            return Ok(Scalar::from_i32(-1));
+            throw_unsup_format!(
+                "eventfd: Flags other than EFD_CLOEXEC, EFD_NONBLOCK \
+            and EFD_SEMAPHORE is not supported."
```
Comment (RalfJung) [inline src/shims/unix/linux/eventfd.rs:182]: EFD_SEMAPHORE is also not supported...
```suggestion
                "eventfd: encountered unknown unsupported flags {:#x}"
```

File: src/shims/unix/linux/eventfd.rs (hunk 3 above)
Comment (tiif) [reply src/shims/unix/linux/eventfd.rs:67]: Ah yes, thanks for catching this. It is likely that my eyes parse the word ``read`` and decided to put it in ``read`` function ><

File: src/shims/unix/linux/eventfd.rs (hunk 5 above)
Comment (tiif) [reply src/shims/unix/linux/eventfd.rs:182]: Oh yea this gave the illusion of we actually support EFD_SEMAPHORE when in fact, they will hit the unsup EFD_SEMAPHORE branch.

**Comment by RalfJung** [comment]
Looks good, thanks. :)
There's no need to rebase when there are no conflicts, and github is terrible at dealing with foce-pushes, so please only do that when needed.

Please squash.

**Comment by RalfJung** [comment]
Hm, that force-push has a non-empty diff? Did you do squash+rebase?

**Comment by RalfJung** [comment]
PR looks the same though.
For reviewing it's easier if you don't do rebase unless needed -- including for the final squash.
(One day maybe github can do git-range-diff, but until then we'll all spend a lot of time working around this...)

@bors r+

**Comment by tiif** [comment]
Ah I see. I did rebase on upstream before squash, will prevent doing that next time.

**Pull request merged**